Tests/Opc.Ua.Security.Certificates.Tests/ExtensionTests.cs

```diff
@@ -0,0 +1,208 @@
+/* ========================================================================
+ * Copyright (c) 2005-2018 The OPC Foundation, Inc. All rights reserved.
+ *
+ * OPC Foundation MIT License 1.00
+ * 
+ * Permission is hereby granted, free of charge, to any person
+ * obtaining a copy of this software and associated documentation
+ * files (the "Software"), to deal in the Software without
+ * restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following
+ * conditions:
+ * 
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+ * OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+ * WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ *
+ * The complete license agreement can be found here:
+ * http://opcfoundation.org/License/MIT/1.00/
+ * ======================================================================*/
+
+
+using System.IO;
+using System.Numerics;
+using System.Security.Cryptography.X509Certificates;
+using NUnit.Framework;
+
+namespace Opc.Ua.Security.Certificates.Tests
+{
+    /// <summary>
+    /// Tests for the CertificateFactory class.
+    /// </summary>
+    [TestFixture, Category("X509Extensions")]
+    [Parallelizable]
+    [SetCulture("en-us")]
+    public class ExtensionTests
+    {
+        #region DataPointSources
+        [DatapointSource]
+        public CertificateAsset[] CertificateTestCases = new AssetCollection<CertificateAsset>(TestUtils.EnumerateTestAssets("*.?er")).ToArray();
+        #endregion
+
+        #region Test Methods
+        [Theory]
+        public void DecodeExtensions(
+            CertificateAsset certAsset
+            )
+        {
+            var x509Cert = new X509Certificate2(certAsset.Cert);
+            Assert.NotNull(x509Cert);
+            TestContext.Out.WriteLine($"CertificateAsset:");
+            TestContext.Out.WriteLine(x509Cert);
+            var altName = X509Extensions.FindExtension<X509SubjectAltNameExtension>(x509Cert);
```

Comment:
I would expect Asserts to check the extensions, not only write them into test output

Reply:
Since these are certs from the test assets, the idea is just to catch an unhandled crash during decode and to compare the output of 'native extension decoder' and the new ones..